package/MDAnalysis/analysis/msd.py

```diff
@@ -421,3 +456,46 @@ def _conclude_fft(self):  # with FFT, np.float64 bit prescision required.
                 positions[:, n, :]
             )
         self.results.timeseries = self.results.msds_by_particle.mean(axis=1)
+    
+
+    def _conclude_modified(self):
+        from collections import defaultdict
+        print("The Dump file has non-linear time spacing")
+        
+        dump_times = [ts.time for ts in self._trajectory]
+        # print(dump_times)
+        n_frames = len(dump_times)
+        n_atoms = self.n_particles
+        positions = np.zeros((n_frames, n_atoms, 3))
+        positions = self._position_array.astype(np.float64)
```

Comment:
This overwrites `positions`. Delete the line above?? Not sure what the intent was here.